src/Symfony/Component/ExpressionLanguage/ExpressionFunction.php

```diff
@@ -42,20 +42,29 @@ class ExpressionFunction
     public function __construct(string $name, callable $compiler, callable $evaluator)
     {
         $this->name = $name;
-        $this->compiler = $compiler;
-        $this->evaluator = $evaluator;
+        $this->compiler = $compiler instanceof \Closure ? $compiler : \Closure::fromCallable($compiler);
+        $this->evaluator = $evaluator instanceof \Closure ? $evaluator : \Closure::fromCallable($evaluator);
```

Comment:
We might not even need the tenary, WDYT?

https://3v4l.org/bMkFh

Reply:
```suggestion
        $this->compiler = \Closure::fromCallable($compiler);
        $this->evaluator = \Closure::fromCallable($evaluator);
```

Reply:
We don't need to, although in perf critical code paths that can make a difference.